java/viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.java

```diff
@@ -55,176 +57,185 @@
+     *
+     * @param featureCollectionTable containing the observation point.
+     */
+    private void performGeoprocessing(final FeatureCollectionTable featureCollectionTable) {
+        // geoprocessing parameters
+        final ListenableFuture<GeoprocessingParameters> parameterFuture = mGeoprocessingTask.createDefaultParametersAsync();
+        parameterFuture.addDoneListener(() -> {
+            try {
+                GeoprocessingParameters parameters = parameterFuture.get();
+                parameters.setProcessSpatialReference(featureCollectionTable.getSpatialReference());
+                parameters.setOutputSpatialReference(featureCollectionTable.getSpatialReference());
+
+                // use the feature collection table to create the required GeoprocessingFeatures input
+                parameters.getInputs().put("Input_Observation_Point", new GeoprocessingFeatures(featureCollectionTable));
+
+                // initialize job from mGeoprocessingTask
+                mGeoprocessingJob = mGeoprocessingTask.createJob(parameters);
+
+                // start the job
+                mGeoprocessingJob.start();
+
+                // listen for job success
+                mGeoprocessingJob.addJobDoneListener(new Runnable() {
+                    @Override
+                    public void run() {
+                        // hide the LoadingView when job is done loading
+                        mLoadingView.setVisibility(View.GONE);
+                        if (mGeoprocessingJob.getStatus() == Job.Status.SUCCEEDED) {
+                            GeoprocessingResult geoprocessingResult = mGeoprocessingJob.getResult();
+                            // get the viewshed from geoprocessingResult
+                            GeoprocessingFeatures resultFeatures = (GeoprocessingFeatures) geoprocessingResult.getOutputs()
+                                    .get("Viewshed_Result");
+                            FeatureSet featureSet = resultFeatures.getFeatures();
+                            for (Feature feature : featureSet) {
+                                Graphic graphic = new Graphic(feature.getGeometry());
+                                mResultGraphicsOverlay.getGraphics().add(graphic);
+                            }
+                        } else {
+                            Toast.makeText(getApplicationContext(), "Geoprocessing result failed!", Toast.LENGTH_LONG).show();
+                        }
+                    }
+                });
+            } catch (InterruptedException | ExecutionException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        mMapView.pause();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        mMapView.resume();
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        mMapView.dispose();
+    }
```

Comment:
The order here should match the life cycles here

https://devtopia.esri.com/runtime/common-samples/wiki/Android-Samples-Repo%3A-A-Contribution-Guide#error-handling